src/test/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactoryTest.java

```diff
@@ -0,0 +1,147 @@
+package io.github.oliviercailloux.y2018.apartments.apartment;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Test class for ApartmentFactory.
+ *
+ * @author Gabriel GUISSET & Clémence COUSIN
+ */
+class ApartmentFactoryTest {
+
+	//Logger for AparmentFactoryTest class
+	private static final Logger LOGGER = LoggerFactory.getLogger(ApartmentFactoryTest.class);
+
+	/**
+	 * Test GenerateApartment function 
+	 * tests the correspondence between the results sent and those recorded.
+	 *
+	 * @author Gabriel GUISSET & Clémence COUSIN
+	 */
+	@Test
+	void testGenerateApartmentFromParameters() {
+		double floorArea = 456.56; 
+		String address = "A Random Address";
+		int nbBedrooms = 5;
+		int nbSleeping = 9;
+		int nbBathrooms = 3;
+		boolean terrace = true;
+		double floorAreaTerrace = 25.32; 
+		String description = "A Random Description for A Random House in A Random Address"; 
+		String title = "A Random House"; 
+		boolean wifi = true;
+		double pricePerNight = 45.95;
+		int nbMinNight = 60; 
+		boolean tele = false;
+		Apartment apart = ApartmentFactory.generateApartment(floorArea, address, nbBedrooms,
+				nbSleeping, nbBathrooms, terrace, floorAreaTerrace, description, 
+				title, wifi, pricePerNight, nbMinNight, tele);
+
+		assertEquals(floorArea, apart.getFloorArea());
+		assertEquals(address, apart.getAddress());
+		assertEquals(nbBedrooms, apart.getNbBedrooms());
+		assertEquals(nbSleeping, apart.getNbSleeping());
+		assertEquals(nbBathrooms, apart.getNbBathrooms());
+		assertEquals(terrace, apart.getTerrace());
+		assertEquals(floorAreaTerrace, apart.getFloorAreaTerrace());
+		assertEquals(description, apart.getDescription());
+		assertEquals(title, apart.getTitle());
+		assertEquals(wifi, apart.getWifi());
+		assertEquals(pricePerNight, apart.getPricePerNight());
+		assertEquals(nbMinNight, apart.getNbMinNight());
+		assertEquals(tele, apart.getTele());
+	}
+
+	/**
+	 * this function tests the immutability of an instance of Apartment using
+	 * Method invoke to check the accessibility of setters.
+	 */
+	@Test
+	void testImmutableApartment() {		
+		Apartment apart = ApartmentFactory.generateRandomApartment();
+		//Test boolean Setters Permissions
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setWifi", boolean.class).invoke(apart, true);  });
```

Comment:
C’est pousser le bouchon un peu loin…

Reply:
Est ce que nous devons selon vous supprimer les tests d'immuabilité ?

Reply:
fixed in commit 8311088296f1ce053dfcd8bfaae15f30945b2afc